crates/memory-testing/cases.json

```diff
@@ -80,8 +80,7 @@
       "memory_lookups": [
         {
           "name": "Key",
-          "hex": "3bc0520a0abff0097d521ce0ee5e5b1cee301939a84742623c0c1697d7a4bd46",
-          "allowed_count": 3
```

Comment:
Should we remove support for `allowed_count`?

Reply:
I'm okay removing it, but it's only two lines of code so I don't think it's a problem either way. 

That said now with the custom allocator hopefully we won't need to allow many leaks.